data/mods/ssb/moves.ts

```diff
@@ -907,6 +907,24 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 				}
 				return 5;
 			},
+			onModifyMove(move, attacker, defender) {
+				if (move.useBaseOffensiveStatAndBoosts &&
+					(!move.useBaseOffensiveStatAndBoosts.includes('atk') && !move.useBaseOffensiveStatAndBoosts.includes('spa'))) return;
+				if (move.useBaseOffensiveStatAndBoosts && move.useBaseOffensiveStatAndBoosts.includes('target')) {
+					if (!defender) return;
+					if (defender.getStat('atk') > defender.getStat('spa')) {
+						move.useBaseOffensiveStatAndBoosts = 'target:spa';
+					} else {
+						move.useBaseOffensiveStatAndBoosts = 'target:atk';
+					}
+				} else {
+					if (attacker.getStat('atk') > attacker.getStat('spa')) {
+						move.useBaseOffensiveStatAndBoosts = 'source:spa';
+					} else {
+						move.useBaseOffensiveStatAndBoosts = 'source:atk';
+					}
+				}
```

Comment:
optional chaining, and using arrays this could be further simplified

Reply:
Done